universal_transfer_operator/src/universal_transfer_operator/data_providers/filesystem/base.py

```diff
@@ -119,7 +119,7 @@ def write(self, source_ref: FileStream):
 
     def write_using_smart_open(self, source_ref: FileStream):
         """Write the source data from remote object i/o buffer to the dataset using smart open"""
-        mode = "wb" if self.read_as_binary(source_ref.actual_filename) else "w"
+        mode = "wb" if self.read_as_binary(str(source_ref.actual_filename)) else "w"
```

Comment:
I'm not a fan of us casting in the middle of a statement like this. Why isn't actual_filename a string?